server/src/main/java/org/opensearch/common/cache/store/config/StoreAwareCacheConfig.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.store.config;
+
+import org.opensearch.common.cache.store.listeners.StoreAwareCacheEventListener;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+
+import java.util.function.ToLongBiFunction;
+
+/**
+ * Configurations related to store aware caches
+ *
+ * @opensearch.internal
+ */
+public class StoreAwareCacheConfig<K, V> {
+
+    private long maxWeightInBytes;
+
+    private ToLongBiFunction<K, V> weigher;
+
+    private TimeValue expireAfterAcess;
+
+    private StoreAwareCacheEventListener<K, V> eventListener;
+
+    private Settings settings;
+
+    private Class<K> keyType;
+
+    private Class<V> valueType;
+
+    private String storagePath;
+
+    private String threadPoolAlias;
+
+    private String cacheAlias;
+
+    private String settingPrefix;
+
+    // Provides capability to make event listener to run in sync/async mode.
+    private boolean isEventListenerModeSync;
+
+    private StoreAwareCacheConfig(Builder<K, V> builder) {
+        this.cacheAlias = builder.cacheAlias;
+        this.keyType = builder.keyType;
+    }
```

Comment:
Should all the fields be `final` and read from the builder?

Reply:
Yeah. I missed this. Will fix.